test/client_test.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import ray
+import sys
+import pytest
+
+
+# NOTE: These client tests run a gateway locally, which I don't think
+# makes any sense in any real-world scenario. A true Client test would
+# require multiple machines with a network configured more realistically.
```

Comment:
One way to do the test on a single machine would be to first start the Ray cluster with

```
ray start --head --with-gateway ...
```

probably inside of `subprocess.Popen`, and then connect here via

```python
ray.init(gateway_address=...)
```

Some examples of this kind of thing are in e.g., https://github.com/ray-project/ray/blob/46351957bb633b1860b4e379e592e2cd9455600a/test/multi_node_test.py#L32 though note that it's ok to defer this kind of test to a subsequent PR.